### PR TITLE
Split package in 2

### LINKED ISSUE #9: Extract React rules into its own package
It's currently a bit annoying to have to install `eslint-plugin-react` (or live with ugly warning from `npm>=3`) even for projects not using React.

Would it be possible to split up this package like airbnb (https://github.com/airbnb/javascript/tree/master/packages) has? One called `base` (or something), which is all current rules except react, and then the base one extends that, plus react rules (so remains the same as the current one).

I can make the code changes if you want

### PR DESCRIPTION
**Pull request by SimenB**
All original rules except React rules are in `base`, while the old
package extends `base` and adds the react rules.

The advantage here is that unless you opt in to use react, you don't
have to install `eslint-plugin-react`

Fixes #9 

//cc @ivarconr @sveisvei @gregersrygg

### CODE REVIEW

**Comment by SimenB** [comment]
Deleted the symlink, same approach as https://github.com/airbnb/javascript/commit/81d01645c2445b67ecd4aa22bd4c5620df43e959

File: .editorconfig [hunk 1]
```diff
@@ -10,5 +10,5 @@ trim_trailing_whitespace = true
 [*.md]
 trim_trailing_whitespace = false
 
-[{*.json,.eslintrc,.babelrc}]
-indent_size = 2
\ No newline at end of file
+[*.json]
```
Comment (SimenB) [inline .editorconfig:13]: `.babelrc` doesn't exist here, and existing `.eslintrc` hade a mix of 2 og and 4 space indentation

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/node_modules/
+node_modules/
```
Comment (SimenB) [inline .gitignore:1]: ignores nested `node_modules`

File: packages/eslint-config-spt-base/package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "eslint-config-spt-base",
+  "version": "0.0.1",
```
Comment (SimenB) [inline packages/eslint-config-spt-base/package.json]: This should be bumped to 1.0 before you publish it 😄

**Comment by SimenB** [comment]
Aight, should be good now.

It'll fail on travis as there are no published versions of `eslint-config-spt-base`, but you can checkout this banch locally, remove `base` from `dependencies` in normal config, and `npm it` should pass 😄

**Comment by ivarconr** [comment]
is the `eslint-config-spt` only intended to add react stuff? Could it be named `eslint-config-spt-react` instad? What happens if some wants to use something else? Angular2, polymer/webcomponents?

**Comment by SimenB** [comment]
Fair point! It's a breaking change though. 

And coming from airbnb, the thought is that it's every rule they have, not react specific.

**Comment by SimenB** [comment]
@nosilleg @Andersos Thoughts here?

File: .gitignore (hunk 2 above)
Comment (nosilleg) [inline .gitignore:1]: Wouldn't all nested node_modules be in the ignored node_modules folder?

**Comment by nosilleg** [comment]
I like the concept, have had a quick read through, but will have another tonight when not on phone. Will need some npm setup from @Andersos if merged.

File: .editorconfig (hunk 1 above)
Comment (Andersos) [inline .editorconfig:13]: We should probably reindent the `.eslintrc` file to make sure it is following the standard then.

**Comment by Andersos** [comment]
So If I understand it correctly we will have two packages with this change. `spt-base` that cointains everything except the React.js rules and `spt` that extends the base and adds the React.js rules.

(1) Is the idea that people that dont use React.js will only use the `spt-base` package then? 
(2) The `spt-style` is just for utility and won't be released?
(3) We probably have to update the [publishing steps](https://github.com/schibsted/eslint-config-spt#publishing) in the README but I can add that if you don't want to @SimenB. 

If I get to know your npm username @nosilleg I can give you permission to the package on npm or I could share with you access to [schibstedbot](https://www.npmjs.com/~schibstedbot).

**Comment by SimenB** [comment]
@Andersos That's right.

1: Yup. The main purpose is to avoid the react plugin dep for projects not using react
2: Yeah, just a meta package. AirBnB have published it for the (now outdated) README, but there is no config in it. No real need to push it, IMO
3: That's true! I can give it a whack later today 😄

File: .gitignore (hunk 2 above)
Comment (SimenB) [inline .gitignore:1]: There are separate `node_modules` in the modules in `packages`

File: .editorconfig (hunk 1 above)
Comment (SimenB) [inline .editorconfig:13]: I have indented it correctly now (4)

Comment (nosilleg) [inline .editorconfig:13]: So we're treating .eslintrc different from other json files? (as per next line, the indent for json is 4)

File: .gitignore (hunk 2 above)
Comment (nosilleg) [inline .gitignore:1]: I'm still not following.  Can you provide an example path, please?  To my understanding there shouldn't be any `node_modules` that aren't under the root `node_modules`

**Comment by nosilleg** [comment]
Having had some time to ponder this, I'm not sure that making react 'de facto' is the right approach.  I think the simpler named `eslint-config-spt` should be the base, and there should be an additional package called 'eslint-config-spt-env-react' that has the react specific stuff in it.  (Essentially flipping the work you've done here.)

The net result is that `env-react.js` becomes it's own package and the inclusion changes from

``` json
"extends": [
    "spt",      
    "spt/env-react"     
]
```

to

``` json
"extends": [
    "spt",      
    "spt-env-react"     
]
```

That removes the need to include react as a dependency when you're not using it, and keeps the main package about linting javascript, not a particular framework.

The biggest downside of this for me is that there will be some `spt/env-*` and one `spt-env-*`.  A subtle thing that I suspect will catch out a few people.

Thoughts?

**Comment by nosilleg** [comment]
@Andersos npm account setup, same username as here.  :smile: Thanks.

**Comment by SimenB** [comment]
@nosilleg Yeah, that could work as well. I went the route I did because it's what airbnb does, but we of course don't have to copy that approach.

What about calling it just `spt-react`, dropping then `env` part?

**Comment by nosilleg** [comment]
Short is nice, but...  What about the other env names? Cut those as well? I tend to fall on the verbosity isnt bad side of things, so my natural reaction is to keep the env.

Having said that, I'll submit to popular opinion without grudge.

**Comment by Andersos** [comment]
I agree that the base version should not include React.js and that is something we should actively choose.

`spt-react` over `spt-env-react`.

**Comment by SimenB** [comment]
@Andersos Updated now, so it's `eslint-config-spt` and `eslint-config-spt-react`.

Which is a breaking change, unlike the last one

File: .editorconfig (hunk 1 above)
Comment (SimenB) [inline .editorconfig:13]: All files are now indented at 4 except for json files (in practice just `package.json`, which npm rewrites into 2-space indent)

File: .gitignore (hunk 2 above)
Comment (SimenB) [inline .gitignore:1]: both `packages/eslint-config-spt/node_modules` and `packages/eslint-config-spt-react/node_modules` exists. Top level have no `node_modules` except for the linked `eslint-config-spt`

File: packages/eslint-config-spt-react/package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "eslint-config-spt-react",
+  "version": "0.0.1",
```
Comment (SimenB) [inline packages/eslint-config-spt-react/package.json:3]: remember to bump this

**Comment by Andersos** [comment]
We could probably release a beta of this and pull inn Eslint@3. I'm still on vacation and will come back next week. Probably gonna take a stab at it then unless someone else already had the time =) Thanks for your work on this @SimenB.

**Comment by SimenB** [comment]
@Andersos Any news here? Sorry to nag on about this, but it's turning into a bottleneck for us.

**Comment by nosilleg** [comment]
@SimenB There's now a v3.0.0 branch that you can target your PRs at.  I'll merge them into there and we can prepare for a v3 release.

**Pull request closed without merging**